backend/app/notify/webhook.go

```diff
@@ -0,0 +1,142 @@
+	"text/template"
+	"time"
+
+	log "github.com/go-pkgz/lgr"
+	"github.com/pkg/errors"
+)
+
+const (
+	webhookTimeOut  = 5 * time.Second
+	webhookTemplate = `{"text": "{{.Text}}"}`
+)
+
+// WebhookParams contain settings for webhook notifications
+type WebhookParams struct {
+	WebhookURL string
+	Template   string
+	Headers    string
+	Timeout    time.Duration
+}
+
+// webhookHeaders is a map where key is an HTTP header name, and value is a list of header values
+type webhookHeaders map[string][]string
+
+// Webhook implements notify.Destination for Webhook notifications
+type Webhook struct {
+	WebhookParams
+	webhookHeaders
+	webhookTemplate *template.Template
+}
+
+// NewWebhook makes Webhook
+func NewWebhook(params WebhookParams) (*Webhook, error) {
+	res := &Webhook{WebhookParams: params}
+	if res.WebhookURL == "" {
+		return nil, errors.New("webhook webhook URL is required for webhook notifications")
+	}
+
+	if res.Template == "" {
+		res.Template = webhookTemplate
+	}
+
+	payloadTmpl, err := template.New("webhook").Parse(res.Template)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to parse webhook template")
+	}
+
+	res.webhookTemplate = payloadTmpl
+	res.webhookHeaders = buildHeaders(res.Headers)
+
+	if res.Timeout == 0 {
+		res.Timeout = webhookTimeOut
+	}
+
+	log.Printf("[DEBUG] create new Webhook notifier for %s", res.WebhookURL)
+
+	return res, nil
+}
+
+// Send sends Webhook notification
+func (t *Webhook) Send(ctx context.Context, req Request) error {
+	var payload bytes.Buffer
+	err := t.webhookTemplate.Execute(&payload, req.Comment)
+	if err != nil {
+		return errors.Wrap(err, "unable to compile webhook template")
+	}
+
+	err = t.doSend(ctx, &payload)
+	if err != nil {
+		return errors.Wrap(err, "webhook notification failed")
+	}
+
+	log.Printf("[DEBUG] send webhook notification, comment id %s", req.Comment.ID)
+
+	return nil
+}
+
+func (t *Webhook) doSend(ctx context.Context, payload io.Reader) error {
+	req, err := http.NewRequestWithContext(ctx, "POST", t.WebhookURL, payload)
+	if err != nil {
+		return errors.Wrap(err, "unable to create webhook request")
+	}
+
+	for header, values := range t.webhookHeaders {
+		for _, value := range values {
+			req.Header.Add(header, value)
+		}
+	}
+
+	client := http.Client{Timeout: t.Timeout}
+	resp, err := client.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "webhook request failed")
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		respBody, _ := ioutil.ReadAll(resp.Body)
+		return fmt.Errorf("webhook request failed with non-OK status code: %d, body: %s", resp.StatusCode, respBody)
+	}
+
+	return nil
+}
+
+// buildHeaders parses parameter string and creates a map of HTTP headers,
+// where key is an HTTP header name, and value is a list of header values
+// For example parameter string Header1:Value1,Header1:Value2,Header2:Value3
+// will produce a map {Header1: [Value1, Value2], Header2: [Value3]}
+func buildHeaders(opt string) webhookHeaders {
```

Comment:
I don't think it will survive a complex headers, like this one 

Access-Control-Allow-Headers:"DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type"

see the similar issue here https://github.com/umputun/reproxy/issues/100 and the solution proposed and implemented

Reply:
Let's go with `splitAtCommas` instead